### PR TITLE
Fixed in-cluster `apinetClient`

### PR DESCRIPTION
**Pull request by nitrocb**
# Proposed Changes
- added in-cluster case for apinetCfg, as this was not covered before leading to a panic (nilpointer-deref) for in-cluster cases

### CODE REVIEW

File: cmd/libvirt-provider/app/app.go [hunk 1]
```diff
@@ -222,20 +223,28 @@ func Run(ctx context.Context, opts Options) error {
 
 	// Check if apinetKubeconfig is provided
 	var apinetClient client.Client
```
Comment (lukasfrank) [inline cmd/libvirt-provider/app/app.go:225]: Do we still need this variable here?

File: cmd/libvirt-provider/app/app.go [hunk 2]
```diff
@@ -222,20 +223,28 @@ func Run(ctx context.Context, opts Options) error {
...
+		// assuming in-cluster config
+		apinetCfg, err = rest.InClusterConfig()
 		if err != nil {
-			setupLog.Error(err, "failed to initialize api-net client")
+			setupLog.Error(err, "failed to create in-cluster-config")
 			return err
 		}
 	}
 
+	apinetClient, err = client.New(apinetCfg, client.Options{Scheme: scheme})
```
Comment (lukasfrank) [inline cmd/libvirt-provider/app/app.go:242]:
```suggestion
	apinetClient, err := client.New(apinetCfg, client.Options{Scheme: scheme})
```

File: cmd/libvirt-provider/app/app.go (hunk 1 above)
Comment (nitrocb) [reply cmd/libvirt-provider/app/app.go]: As the client creation is not part of the if condition anymore, we don't need that variable anymore, too.
Thanks for that hint! I have removed it.

**Review by lukasfrank — APPROVED** [review]
Thanks a lot for your contribution! LGTM

**Pull request merged**